src/components/create-guild/MultiPlatformGrid/components/MultiPlatformSelectButton.tsx

```diff
@@ -0,0 +1,207 @@
+import CreateGuildDiscord from "components/create-guild/MultiPlatformGrid/components/CreateGuildDiscord"
+import CreateGuildGithub from "components/create-guild/MultiPlatformGrid/components/CreateGuildGithub"
+import CreateGuildGoogle from "components/create-guild/MultiPlatformGrid/components/CreateGuildGoogle"
+import CreateGuildSecretText from "components/create-guild/MultiPlatformGrid/components/CreateGuildSecretText"
+import CreateGuildTelegram from "components/create-guild/MultiPlatformGrid/components/CreateGuildTelegram"
+import CreateGuildTwitter from "components/create-guild/MultiPlatformGrid/components/CreateGuildTwitter"
+import CreateGuildUniqueText from "components/create-guild/MultiPlatformGrid/components/CreateGuildUniqueText"
+import Image from "next/image"
+import { CheckCircle, IconProps } from "phosphor-react"
+import platforms from "platforms/platforms"
+import { ComponentType, RefAttributes } from "react"
+import { useFormContext, useWatch } from "react-hook-form"
+import { GuildFormType, PlatformName, Rest } from "types"
+import { useAccount } from "wagmi"
+
+export type PlatformHookType = ({
+  platform,
+  onSelection,
+}: {
+  platform: PlatformName
+  onSelection: (platform: PlatformName) => void
+}) => {
+  onClick: () => void
+  isLoading: boolean
+  loadingText: string
+  rightIcon: ComponentType<IconProps & RefAttributes<SVGSVGElement>>
+}
+
+const createGuildPlatformComponents: Record<
+  Exclude<PlatformName, "POAP" | "TWITTER_V1" | "EMAIL">,
+  (props: { isOpen: boolean; onClose: () => void }) => JSX.Element
+> = {
+  DISCORD: CreateGuildDiscord,
+  TELEGRAM: CreateGuildTelegram,
+  GOOGLE: CreateGuildGoogle,
+  GITHUB: CreateGuildGithub,
+  CONTRACT_CALL: CreateGuildContractCall,
+  TEXT: CreateGuildSecretText,
+  TWITTER: CreateGuildTwitter,
+  UNIQUE_TEXT: CreateGuildUniqueText,
+}
+
+type Props = {
+  platform: PlatformName
+  hook?: PlatformHookType
+  title: string
+  description?: string
+  imageUrl?: string
+  icon?: ComponentType<IconProps & RefAttributes<SVGSVGElement>>
+  onSelection: (platform: PlatformName) => void
+} & Rest
+
+const MultiPlatformSelectButton = ({
+  platform,
+  title,
+  description,
+  imageUrl,
+  icon,
+  onSelection,
+  ...rest
+}: Props) => {
+  const { address } = useAccount()
+  const { openWalletSelectorModal } = useWeb3ConnectionManager()
+  const methods = useFormContext()
+  const { setValue } = useFormContext<GuildFormType>()
+  const { isOpen, onClose, onOpen } = useDisclosure()
+  const user = useUser()
+
+  const { onConnect, isLoading, loadingText } = useConnectPlatform(
+    platform,
+    () => onSelection(platform),
+    false,
+    "creation"
+  )
+
+  const guildPlatforms = useWatch({ name: "guildPlatforms" })
+  const twitterLink = useWatch({ name: "socialLinks.TWITTER" })
+
+  const removePlatform = (platformName: PlatformName) => {
+    const guildPlatformsCopy = JSON.parse(JSON.stringify(guildPlatforms))
```

Comment:
I think we don't need a deep copy of guildPlatform here, since we don't modify it